### PR TITLE
[Flink AI Flow] Metric center refactor

### PR DESCRIPTION
**Pull request by SteNicholas**
The service of Metric Center doesn't support the need of online pipeline metric. It should be refactored through refactoring the data structure of the Metric model.

### CODE REVIEW

File: flink-ai-flow/ai_flow/metric/service/metric_service.py [hunk 1]
```diff
@@ -16,144 +16,131 @@
...
+    def updateMetricSummary(self, request, context):
+        metric_summary_proto = request.metric_summary
+        metric_summary = proto_to_metric_summary(metric_summary_proto)
+        res_metric_summary = self.store.update_metric_summary(metric_summary.uuid,
+                                                              metric_summary.metric_name,
+                                                              metric_summary.metric_key,
+                                                              metric_summary.metric_value,
+                                                              metric_summary.metric_timestamp,
+                                                              metric_summary.model_version,
+                                                              metric_summary.job_execution_id)
```
Comment (jiangxin369) [inline flink-ai-flow/ai_flow/metric/service/metric_service.py:119]: Does user know `job_execution_id` when they update `MetricSummary` ?

File: flink-ai-flow/ai_flow/store/sqlalchemy_store.py [hunk 2]
```diff
@@ -1358,290 +1358,131 @@ def get_model_version_detail(self, model_version):
...
-        register metric summary
-        :param uuid: metric summary unique id
-        :param metric_id: associate with metric meta uuid
-        :param metric_key:
-        :param metric_value:
-        :return:
-        """
+                raise AIFlowException('Update metric meta failed! Error: {}.'.format(str(e)))
+
+    def delete_metric_meta(self, metric_name):
```
Comment (jiangxin369) [inline flink-ai-flow/ai_flow/store/sqlalchemy_store.py:1422]: Does metric summary should be deleted in a cascade?

File: flink-ai-flow/ai_flow/store/db/db_model.py [hunk 3]
```diff
@@ -306,13 +298,19 @@ class SqlMetricSummary(base, Base):
     SQL model of metric summary
     """
     __tablename__ = 'metric_summary'
-    metric_id = Column(BigInteger, nullable=False)
```
[the remark this answers is not in the document]
Comment (SteNicholas) [reply flink-ai-flow/ai_flow/store/db/db_model.py:309]: In `SqlMetricSummary`, there is no foreign key. Just use `metric_name` associate with the `SqlMetricMeta`.

File: flink-ai-flow/ai_flow/store/sqlalchemy_store.py [hunk 4]
```diff
@@ -1653,34 +1494,109 @@ def get_model_metric_meta(self, model_name, model_version) -> Union[None, Metric
...
+
+    def delete_metric_summary(self, uuid: int):
+        with self.ManagedSessionMaker() as session:
+            try:
+                conditions = [
+                    SqlMetricSummary.uuid == uuid
+                ]
+                metric_summary_table = session.query(SqlMetricSummary).filter(*conditions).first()
+                metric_summary_table.is_deleted = TRUE
+                session.add(metric_summary_table)
```
[the remark this answers is not in the document]
Comment (SteNicholas) [reply flink-ai-flow/ai_flow/store/sqlalchemy_store.py:1550]: `session.add` isn't same as `session.commit`. Therefore here need this operation. You could refer to `_save_to_db` method.

File: flink-ai-flow/ai_flow/metric/service/metric_service.py (hunk 1 above)
Comment (SteNicholas) [reply flink-ai-flow/ai_flow/metric/service/metric_service.py:119]: `job_execution_id ` is known when updating summary for the online pipeline of ODL,

**Review by jiangxin369 — APPROVED** [review]

**Pull request merged**